gateway/ln-gateway/src/lnd.rs

```diff
@@ -325,27 +351,51 @@ impl ILnRpcClient for GatewayLndClient {
             self.macaroon.clone(),
         )
         .await?;
-        let send_response = client
-            .lightning()
-            .send_payment_sync(SendRequest {
-                payment_request: invoice.invoice.to_string(),
-                ..Default::default()
-            })
-            .await
-            .map_err(|e| anyhow::anyhow!(format!("LND error: {e:?}")))?
-            .into_inner();
-        info!("send response {:?}", send_response);
 
-        if send_response.payment_preimage.is_empty() {
-            return Err(GatewayError::LnRpcError(tonic::Status::new(
-                tonic::Code::Internal,
-                "LND did not return a preimage",
-            )));
+        // If the payment exists, that means we've already tried to pay the invoice
+        let preimage = if let Some(mut payment) = self
+            .lookup_payment(invoice.invoice.clone(), invoice.timestamp, &mut client)
+            .await?
+        {
+            while payment.status == 1 {
```

Comment:
This loop sometimes gets stuck for payments. LND will always return an `IN_FLIGHT` status. Haven't figured out why....

Reply:
Would be nice to have a descriptive variable instead of `1` ...

Reply:
Removed